a8c-files.php

```diff
@@ -63,10 +63,16 @@ function __construct() {
 
 	private function init_jetpack_photon_filters() {
 		// The files service has Photon capabilities, but is served from the same domain.
-		// Force Jetpack to use it instead of the default Photon domains (`i*.wp.com`).
-		add_filter( 'jetpack_photon_domain', function() {
-			return home_url();
-		} );
+		// Force Jetpack to use the files service instead of the default Photon domains (`i*.wp.com`) for internal files.
+		// Externally hosted files continue to use the remot Photon service.
+		add_filter( 'jetpack_photon_domain', function( $photon_url, $image_url ) {
+			$home_url = home_url();
+			if ( wp_startswith( $image_url, $home_url ) ) {
```

Comment:
Should we also for *.go-vip.co domains to use the files service in case it's not the home_url? I don't think we'd ever want to send those through Photon.